File: Exemplos/App_eXperience_XamarinAndroid/E1_Xamarin_Android_M8-font/XamarinAndroid-M8-master/Sat/ServiceSat.cs
using Android.Content;
using Android.Widget;
using System.Collections.Generic;
using BR.Com.Elgin;

namespace M8
{
    class ServiceSat
    {
        private readonly Context contextSat;

        public ServiceSat(Context context)
        {
            contextSat = context;
        }

        public string AtivarSAT(Dictionary<string, object> map)
        {
            int numSessao = (int)map["numSessao"];
            int subComando = (int)map["subComando"];
            string codeAtivacao = (string)map["codeAtivacao"];
            string cnpj = (string)map["cnpj"];
            int cUF = (int) map["cUF"];
            string retorno = Sat.AtivarSat(numSessao, subComando, codeAtivacao, cnpj, cUF);
            MostrarRetorno(retorno);
            return retorno;
        }

        public string AssociarAssinatura(Dictionary<string, object> map)
        {
            int numSessao = (int)map["numSessao"];
            string codeAtivacao = (string)map["codeAtivacao"];
            string cnpjSh = (string) map["cnpjSh"];
            string assinaturaAC = (string)map["assinaturaAC"];

            string retorno = Sat.AssociarAssinatura(numSessao, codeAtivacao, cnpjSh, assinaturaAC);
            MostrarRetorno(retorno);
            return retorno;
        }

        public string ConsultarSAT(Dictionary<string, object> map)
        {
            int numSessao = (int) map["numSessao"];

            string result = Sat.ConsultarSat(numSessao);
            MostrarRetorno(result);
            return result;
        }

        public string StatusOperacional(Dictionary<string, object> map)
        {
            int numSessao = (int)map["numSessao"];
            string codeAtivacao = (string)map["codeAtivacao"];

            string retorno = Sat.ConsultarStatusOperacional(numSessao, codeAtivacao);
            MostrarRetorno(retorno);
            return retorno;
        }

        public string EnviarVenda(Dictionary<string, object> map)
        {
            int numSessao = (int) map["numSessao"];
            string codeAtivacao = (string) map["codeAtivacao"];
            string xmlSale = (string) map["xmlSale"];

            string retorno = Sat.EnviarDadosVenda(numSessao, codeAtivacao, xmlSale);
            MostrarRetorno(retorno);
            return retorno;
        }

        public string CancelarVenda(Dictionary<string, object> map)
        {
            int numSessao = (int) map["numSessao"];
            string codeAtivacao = (string)map["codeAtivacao"];
            string cFeNumber = (string)map["cFeNumber"];
            string xmlCancelamento = (string) map["xmlCancelamento"];

            string retorno = Sat.CancelarUltimaVenda(numSessao, codeAtivacao, cFeNumber, xmlCancelamento);
            MostrarRetorno(retorno);
            return retorno;
        }

        private void MostrarRetorno(string retorno)
        {
            Toast.MakeText(contextSat, string.Format("Retorno: {0}", retorno), ToastLength.Long).Show();
        }
    }
}

File: Exemplos/App_eXperience_ReactNative/Elgin_AppExperience_ReactNative-font/android/app/src/main/java/com/elginm8/ServiceSat.java
package com.elginm8;

import android.content.Context;
import android.os.Build;
import android.widget.Toast;

import androidx.annotation.RequiresApi;

import java.util.Arrays;
import java.util.Optional;
import java.util.Random;
import java.util.concurrent.atomic.AtomicReference;

import br.com.elgin.DeviceInfo;
import br.com.elgin.Sat;
import br.com.elgin.SatInitializer;
import com.facebook.react.bridge.ReadableMap;

public class ServiceSat {
    public static SatInitializer satInitializer;
    private static Context contextSat;

    public  ServiceSat (Context context){
        satInitializer = new SatInitializer();
        contextSat = context;
    }

    @RequiresApi(api = Build.VERSION_CODES.N)
    public static String ativarSAT(ReadableMap map){
        int numSessao = getNumeroSessao();
        int subComando = map.getInt("subComando");
        String codeAtivacao = map.getString("codeAtivacao");
        String cnpj = map.getString("cnpj");
        int cUF = map.getInt("cUF");

        StringBuilder valueRetorn = new StringBuilder(Sat.ativarSat(numSessao, subComando, codeAtivacao, cnpj, cUF));
        if (valueRetorn.toString().contains("NomParsing")) {
            valueRetorn = new StringBuilder(valueRetorn.toString().replace("NomParsing(", "").replace(", Digit)", ""));

            int[] arr = Arrays.stream(valueRetorn.substring(1, valueRetorn.length() - 1).split(","))
                    .map(String::trim).mapToInt(Integer::parseInt).toArray();

            valueRetorn = new StringBuilder();
            for (int item : arr) {
                valueRetorn.append((char) item);
            }
        }

        mostrarRetorno(valueRetorn.toString());
        return valueRetorn.toString();
    }

    @RequiresApi(api = Build.VERSION_CODES.N)
    public static String associarAssinatura(ReadableMap map) {
        int numSessao = getNumeroSessao();
        String codeAtivacao = map.getString("codeAtivacao");
        String cnpjSh = map.getString("cnpjSh");
        String assinaturaAC = map.getString("assinaturaAC");

        StringBuilder valueRetorn = new StringBuilder(Sat.associarAssinatura(numSessao, codeAtivacao, cnpjSh, assinaturaAC));
        if (valueRetorn.toString().contains("NomParsing")) {
            valueRetorn = new StringBuilder(valueRetorn.toString().replace("NomParsing(", "").replace(", Digit)", ""));

            int[] arr = Arrays.stream(valueRetorn.substring(1, valueRetorn.length() - 1).split(","))
                    .map(String::trim).mapToInt(Integer::parseInt).toArray();

            valueRetorn = new StringBuilder();
            for (int item : arr) {
                valueRetorn.append((char) item);
            }
        }

        mostrarRetorno(valueRetorn.toString());
        return valueRetorn.toString();
    }

    @RequiresApi(api = Build.VERSION_CODES.N)
    public static String consultarSAT(ReadableMap map) {
        int numSessao = getNumeroSessao();

        StringBuilder valueRetorn = new StringBuilder(Sat.consultarSat((Integer) numSessao));
        if (valueRetorn.toString().contains("NomParsing")) {
            valueRetorn = new StringBuilder(valueRetorn.toString().replace("NomParsing(", "").replace(", Digit)", ""));

            int[] arr = Arrays.stream(valueRetorn.substring(1, valueRetorn.length() - 1).split(","))
                    .map(String::trim).mapToInt(Integer::parseInt).toArray();

            valueRetorn = new StringBuilder();
            for (int item : arr) {
                valueRetorn.append((char) item);
            }
        }

        mostrarRetorno(valueRetorn.toString());
        return valueRetorn.toString();
    }

    @RequiresApi(api = Build.VERSION_CODES.N)
    public static String statusOperacional(ReadableMap map){
        int numSessao = getNumeroSessao();
        String codeAtivacao = map.getString("codeAtivacao");

        StringBuilder valueRetorn = new StringBuilder(Sat.consultarStatusOperacional(numSessao,codeAtivacao));
        if (valueRetorn.toString().contains("NomParsing")) {
            valueRetorn = new StringBuilder(valueRetorn.toString().replace("NomParsing(", "").replace(", Digit)", ""));

            int[] arr = Arrays.stream(valueRetorn.substring(1, valueRetorn.length() - 1).split(","))
                    .map(String::trim).mapToInt(Integer::parseInt).toArray();

            valueRetorn = new StringBuilder();
            for (int item : arr) {
                valueRetorn.append((char) item);
            }
        }

        mostrarRetorno(valueRetorn.toString());

        return valueRetorn.toString();
    }


    @RequiresApi(api = Build.VERSION_CODES.N)
    public static String enviarVenda(ReadableMap map){
        int numSessao = getNumeroSessao();
        String codeAtivacao = map.getString("codeAtivacao");
        String xmlSale = map.getString("xmlSale");

        StringBuilder valueRetorn = new StringBuilder(Sat.enviarDadosVenda(numSessao, codeAtivacao, xmlSale));
        if (valueRetorn.toString().contains("NomParsing")) {
            valueRetorn = new StringBuilder(valueRetorn.toString().replace("NomParsing(", "").replace(", Digit)", ""));

            int[] arr = Arrays.stream(valueRetorn.substring(1, valueRetorn.length() - 1).split(","))
                    .map(String::trim).mapToInt(Integer::parseInt).toArray();

            valueRetorn = new StringBuilder();
            for (int item : arr) {
                valueRetorn.append((char) item);
            }
        }

        mostrarRetorno(valueRetorn.toString());
        return valueRetorn.toString();
    }

    @RequiresApi(api = Build.VERSION_CODES.N)
    public static String cancelarVenda(ReadableMap map){
        int numSessao = getNumeroSessao();
        String codeAtivacao = map.getString("codeAtivacao");
        String cFeNumber = map.getString("cFeNumber");
        String xmlCancelamento = map.getString("xmlCancelamento");

        StringBuilder valueRetorn = new StringBuilder(Sat.cancelarUltimaVenda(numSessao, codeAtivacao, cFeNumber, xmlCancelamento));
        if (valueRetorn.toString().contains("NomParsing")) {
            valueRetorn = new StringBuilder(valueRetorn.toString().replace("NomParsing(", "").replace(", Digit)", ""));

            int[] arr = Arrays.stream(valueRetorn.substring(1, valueRetorn.length() - 1).split(","))
                    .map(String::trim).mapToInt(Integer::parseInt).toArray();

            valueRetorn = new StringBuilder();
            for (int item : arr) {
                valueRetorn.append((char) item);
            }
        }

        mostrarRetorno(valueRetorn.toString());
        return valueRetorn.toString();
    }

    private static void mostrarRetorno(String retorno) {
        Toast.makeText(contextSat, String.format("Retorno: %s", retorno), Toast.LENGTH_LONG).show();
    }

    private static int getNumeroSessao() {
        return new Random().nextInt(1_000_000);
    }
}
